Python/specialize.c

```diff
@@ -1794,6 +1812,32 @@ _Py_Specialize_CompareOp(PyObject *lhs, PyObject *rhs,
             goto success;
         }
     }
+#ifdef Py_STATS
+    if (PyBytes_CheckExact(lhs)) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_BYTES);
+        goto failure;
+    }
+    if (PyTuple_CheckExact(lhs)) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_TUPLE);
+        goto failure;
+    }
+    if (PyList_CheckExact(lhs)) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_LIST);
+        goto failure;
+    }
+    if (PySet_CheckExact(lhs) || PyFrozenSet_CheckExact(lhs)) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_SET);
+        goto failure;
+    }
+    if (PyBool_Check(lhs)) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_BOOL);
+        goto failure;
+    }
+    if (Py_TYPE(lhs)->tp_richcompare == PyBaseObject_Type.tp_richcompare) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_BASEOBJECT);
+        goto failure;
+    }
+#endif
```

Comment:
As above, could you move the classification of the failure into a helper function?
`SPECIALIZATION_FAIL(COMPARE_OP, compare_op_fail_kind(lhs, rhs));`
Otherwise, it obscures the logic of the specialization function.